modules/account-lib/src/coin/dot/transaction.ts

```diff
@@ -1,20 +1,202 @@
+      eraPeriod: decodedTx.eraPeriod,
+      chainName: this._chainName,
+      tip: decodedTx.tip,
+    };
+
+    if (this.type === TransactionType.Send) {
+      const txMethod = decodedTx.method.args as any;
+      if (txMethod.real) {
+        const keypairReal = new KeyPair({
+          pub: Buffer.from(decodeAddress(txMethod.real)).toString('hex'),
+        });
+        result.real = keypairReal.getAddress();
+        result.forceProxyType = txMethod.forceProxyType;
+        const decodedCall = utils.decodeCallMethod(this._dotTransaction, {
+          metadataRpc: this._dotTransaction.metadataRpc,
+          registry: this._registry,
+        });
+        const keypairDest = new KeyPair({
+          pub: Buffer.from(decodeAddress(decodedCall.dest.id)).toString('hex'),
+        });
+        result.to = keypairDest.getAddress();
+        result.amount = decodedCall.value;
+      } else {
+        const keypairDest = new KeyPair({
+          pub: Buffer.from(decodeAddress(txMethod.dest.id)).toString('hex'),
+        });
+        result.to = keypairDest.getAddress();
+        result.amount = txMethod.value;
+      }
+    }
+
+    if (this.type === TransactionType.StakingActivate) {
+      const txMethod = decodedTx.method.args as StakeArgs;
+      const keypair = new KeyPair({
+        pub: Buffer.from(decodeAddress(txMethod.controller.id, false, this._registry.chainSS58)).toString('hex'),
+      });
+
+      result.controller = keypair.getAddress();
+      result.amount = txMethod.value;
+
+      const payee = txMethod.payee as StakeArgsPayeeRaw;
+      if (payee.account) {
+        const keypair = new KeyPair({
+          pub: Buffer.from(decodeAddress(payee.account, false, this._registry.chainSS58)).toString('hex'),
+        });
+        result.payee = keypair.getAddress();
+      } else {
+        const payeeType = utils.capitalizeFirstLetter(Object.keys(payee)[0]) as string;
+        result.payee = payeeType;
+      }
+    }
+
+    if (this.type === TransactionType.WalletInitialization) {
+      const txMethod = decodedTx.method.args as AddProxyArgs;
+      const keypair = new KeyPair({
+        pub: Buffer.from(decodeAddress(txMethod.delegate, false, this._registry.chainSS58)).toString('hex'),
+      });
+      result.owner = keypair.getAddress();
+      result.proxyType = txMethod.proxyType;
+      result.delay = txMethod.delay;
+    }
+
+    if (this.type === TransactionType.StakingUnlock) {
+      const txMethod = decodedTx.method.args as UnstakeArgs;
+      result.amount = txMethod.value;
+    }
+
+    return result;
+  }
+
+  setDotTransaction(tx: UnsignedTransaction): void {
+    this._dotTransaction = tx;
+  }
+
+  getDotTransaction(): UnsignedTransaction | undefined {
+    return this._dotTransaction;
+  }
+
+  setTransactionType(transactionType: TransactionType): void {
+    this._type = transactionType;
+  }
+
+  getTransactionType(): TransactionType | undefined {
+    return this._type;
```

Comment:
Add loadInputAndOutput method in transaction.ts
Example:
https://github.com/BitGo/BitGoJS/blob/66d53d08910bd2f0cf81dc5e21e522419ca1de25/modules/account-lib/src/coin/stx/transaction.ts#L205